servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpServerBuilder.java

```diff
@@ -356,11 +347,13 @@ private HttpExecutionContext buildExecutionContext(final HttpExecutionStrategy s
                 filteredService, drainRequestPayloadBody);
     }
 
-    private HttpExecutionStrategy computeServiceStrategy(Object service) {
-        HttpExecutionStrategy serviceStrategy = requiredOffloads(service, defaultStrategy());
+    private HttpExecutionStrategy computeServiceStrategy(HttpExecutionStrategyInfluencer service) {
+        HttpExecutionStrategy serviceStrategy = service.requiredOffloads();
         HttpExecutionStrategy filterStrategy = computeRequiredStrategy(serviceFilters, serviceStrategy);
-        return defaultStrategy() == strategy ? filterStrategy :
-                strategy.hasOffloads() ? strategy.merge(filterStrategy) : offloadNone();
+        return defaultStrategy() == strategy ?
+            filterStrategy.hasOffloads() ? strategy.merge(filterStrategy) : strategy :
```

Comment:
I am tempted to replace the return of `strategy` when `filterStrategy` has no offloads with `offloadAll()`. Currently this results in `defaultStrategy` appearing the context which will merge differently with non-default execution strategy than `offloadAll()` and potentially result in additional offloading.